script.js

```diff
@@ -1,12 +1,547 @@
+    }
+  }
+
+  showSearch.appendChild(showSearchContainer);
+
+  return showSearchContainer;
+}
+
+
+
+function showListing() {
+  const rootElem = document.getElementById("root");
+  rootElem.innerHTML = "";
+
+  // Create the show container
+  const showContainer = document.createElement("div");
+  showContainer.classList.add("show-container");
+  rootElem.appendChild(showContainer);
+
+  if (episodeSelectContainer) {
+    // Hiding the episode select field if 
+    episodeSelectContainer.style.display = "none";
+  }
+
+  allShows.forEach((show) => {
+    const showCard = createShowCard(show);
+    showContainer.appendChild(showCard);
+  });
+}
+
+//level-500 Display shows
+function createShowCard(show) {
+  // Creating the show card
+  const showCard = document.createElement("div");
+  showCard.classList.add("show-card");
+  
+  const showName = document.createElement("h2");
+  showName.classList.add("show-name");
+  showName.textContent = show.name;
+  showCard.appendChild(showName);
+  
+  const showDetails = document.createElement("div");
+  showDetails.classList.add("show-details");
+  showCard.appendChild(showDetails);
+  
+  // Creating a div to wrap the image and summary
+  const imageSummaryDiv = document.createElement("div");
+  imageSummaryDiv.classList.add("image-summary");
+  showDetails.appendChild(imageSummaryDiv);
+  
+  const showImage = document.createElement("img");
+  showImage.classList.add("show-image");
+  showImage.src = show.image ? show.image.medium : "";
+  showImage.alt = `Poster of ${show.name}`;
+  imageSummaryDiv.appendChild(showImage);
+  
+  const showSummary = document.createElement("p");
+  showSummary.classList.add("show-summary");
+  showSummary.innerHTML = show.summary;
+  imageSummaryDiv.appendChild(showSummary);
+  
+  // Creating a div to wrap the additional details
+  const additionalDetailsDiv = document.createElement("div");
+  additionalDetailsDiv.classList.add("additional-details");
+  showDetails.appendChild(additionalDetailsDiv);
+  
+  const showRated = document.createElement("p");
+  showRated.classList.add("show-rated");
+  showRated.textContent = `Rated: ${show.rating.average}`;
+  additionalDetailsDiv.appendChild(showRated);
+  
+  const showGenres = document.createElement("p");
+  showGenres.classList.add("show-genres");
+  showGenres.textContent = `Genres: ${show.genres.join(", ")}`;
+  additionalDetailsDiv.appendChild(showGenres);
+  
+  const showStatus = document.createElement("p");
+  showStatus.classList.add("show-status");
+  showStatus.textContent = `Status: ${show.status}`;
+  additionalDetailsDiv.appendChild(showStatus);
+  
+  const showRuntime = document.createElement("p");
+  showRuntime.classList.add("show-runtime");
+  showRuntime.textContent = `Runtime: ${show.runtime} minutes`;
+  additionalDetailsDiv.appendChild(showRuntime);
+  
+
+  // Add click event listener to show name -- back to show button TO BE DONE
+  showName.addEventListener("click", () => {
```

Comment:
This listener seems suspicious. Firstly, I would place it at the bottom, for better readability. Secondly, is the promises inside necessary? getEpisodes() is already an async function.